diff --git a/src/PVE/AccessControl.pm b/src/PVE/AccessControl.pm
index 8879ab8..acd9056 100644
--- a/src/PVE/AccessControl.pm
+++ b/src/PVE/AccessControl.pm
@@ -1056,7 +1056,11 @@ sub delete_pool_acl {
 # user: a normal user/customer can to that
 my $privgroups = {
     VM => {
-        root => [],
+        root => [
+            'VM.Config.Nesting',
+            'VM.Config.Keyctl',
+            'VM.Config.Vsock',
+        ],
         admin => [
             'VM.Config.Disk',
             'VM.Config.CPU',
@@ -1092,6 +1096,10 @@ my $privgroups = {
             'Sys.Modify', # edit/change node settings
             'Sys.Incoming', # incoming storage/guest migrations
             'Sys.AccessNetwork', # for, e.g., downloading ISOs from any URL
+            'Sys.ACME.Account.Audit',
+            'Sys.ACME.Account.Create',
+            'Sys.ACME.Account.Modify',
+            'Sys.ACME.Account.Remove',
         ],
         admin => [
             'Sys.Console', 'Sys.Syslog',
@@ -1282,6 +1290,9 @@ sub check_path {
     my ($path) = @_;
     return $path =~ m!^(
 	/
+	|/acme
+	|/acme/accounts
+	|/acme/accounts/[A-Za-z][A-Za-z0-9_\-]+
 	|/access
 	|/access/groups
 	|/access/groups/[[:alnum:]\.\-\_]+
